### PR TITLE
Refactor krnlmon code

### PR DESCRIPTION
**Pull request by nupuruttarwar**
Signed-off-by: nupuruttarwar <nupur.uttarwar@intel.com>

### CODE REVIEW

File: switchlink/sai/switchlink_handle_link.c [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+
+   status = sai_api_query(SAI_API_ROUTER_INTERFACE, (void **)&sai_rintf_api);
+   krnlmon_assert(status == SAI_STATUS_SUCCESS);
+
+  return status;
+}
+
+/*
+ * Routine Description:
+ *    Initialize Virtuiual Router SAI API
```
Comment (ffoulkes) [inline switchlink/sai/switchlink_handle_link.c:43]: typo: should be `Virtual`

File: switchlink/switchlink_db.c [hunk 2]
```diff
@@ -318,7 +318,7 @@ switchlink_db_status_t switchlink_db_tunnel_interface_get_info(
  *    SWITCHLINK_DB_STATUS_ITEM_NOT_FOUND otherwise
  */
 
-switchlink_db_status_t switchlink_db_tunnel_interface_delete(uint32_t ifindex) {
+switchlink_db_status_t switchlink_db_tunnel_delete_interface(uint32_t ifindex) {
```
Comment (ffoulkes) [inline switchlink/switchlink_db.c:321]: The object being deleted is a _tunnel _interface__, so this should be `..._delete_tunnel_interface`.

**Review by ffoulkes — APPROVED** [review]
LGTM

File: switchlink/CMakeLists.txt [hunk 3]
```diff
@@ -5,6 +5,16 @@
 
 cmake_minimum_required(VERSION 3.5)
 
+include(FindPkgConfig)
```
[the remark this answers is not in the document]
Comment (nupuruttarwar) [reply switchlink/CMakeLists.txt:8]: Removed the message since it will be will be duplicate

File: switchlink/sai/switchlink_handle_link.c (hunk 1 above)
Comment (nupuruttarwar) [reply switchlink/sai/switchlink_handle_link.c:43]: Corrected

File: switchlink/switchlink_db.c [hunk 4]
```diff
@@ -237,7 +237,7 @@ switchlink_db_status_t switchlink_db_interface_update(
  *    SWITCHLINK_DB_STATUS_ITEM_NOT_FOUND otherwise
  */
 
-switchlink_db_status_t switchlink_db_interface_delete(uint32_t ifindex) {
+switchlink_db_status_t switchlink_db_delete_interface(uint32_t ifindex) {
```
Comment (n-sandeep) [inline switchlink/switchlink_db.c:240]: Any reason why all the names has been modified?
Following any specific guidelines for function names?

Comment (nupuruttarwar) [reply switchlink/switchlink_db.c:240]: names were changed to be more readable. Guideline is to have a verb first

**Pull request merged**